core/src/main/scala/org/apache/spark/sql/SnappyBaseParser.scala

```diff
@@ -311,12 +311,26 @@ abstract class SnappyBaseParser(session: SparkSession) extends Parser {
     quotedIdentifier
   }
 
+  protected final def packageIdentifierPart: Rule1[String] = rule {
+    atomic(capture((Consts.identifier | Consts.hyphen | Consts.dot).+)) ~ delimiter ~> { (s: String) =>
```

Comment:
delimiter rule is not required I guess. Just ws( whitespaces ) should be fine, otherwise wrong sytaxes will also start getting parsed. Please check.